optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java

```diff
@@ -0,0 +1,722 @@
+                int startMinuteOfDay = 24 * 60, endMinuteOfDay = 0;
+                for (TimeGrain timeGrain : solution.getTimeGrainList()) {
+                    if (timeGrain.getDay().equals(dayOfYear)) {
+                        startMinuteOfDay = timeGrain.getStartingMinuteOfDay() < startMinuteOfDay ?
+                            timeGrain.getStartingMinuteOfDay() : startMinuteOfDay;
+                        endMinuteOfDay = timeGrain.getStartingMinuteOfDay() + TimeGrain.GRAIN_LENGTH_IN_MINUTES > endMinuteOfDay ?
+                            timeGrain.getStartingMinuteOfDay() + TimeGrain.GRAIN_LENGTH_IN_MINUTES : endMinuteOfDay;
+                    }
+                }
+                LocalTime startTime = LocalTime.ofSecondOfDay(startMinuteOfDay * 60); // 8am TODO: set start/end Time fields to class Day
+                LocalTime endTime = LocalTime.ofSecondOfDay(endMinuteOfDay * 60); // 6pm
+                LocalTime lunchHourStartTime = LocalTime.ofSecondOfDay(12 * 60 * 60); // 12 pm
+
+                nextCell().setCellValue(DAY_FORMATTER.format(date));
+                nextCell().setCellValue(TIME_FORMATTER.format(startTime));
+                nextCell().setCellValue(TIME_FORMATTER.format(endTime));
+                nextCell().setCellValue(TIME_FORMATTER.format(lunchHourStartTime));
+            }
+            autoSizeColumnsWithHeader();
+        }
+
+        private void writeRooms() {
+            nextSheet("Rooms", 1, 1, false);
+            nextRow();
+            nextHeaderCell("Name");
+            nextHeaderCell("Capacity");
+            for (Room room : solution.getRoomList()) {
+                nextRow();
+                nextCell().setCellValue(room.getName());
+                nextCell().setCellValue(room.getCapacity());
+            }
+            autoSizeColumnsWithHeader();
+        }
+
+        private void writeRoomsView() {
+            nextSheet("Rooms view", 1, 2, true);
+            String[] filteredConstraintNames = {
+                DO_ALL_MEETINGS_AS_SOON_AS_POSSIBLE,
+
+                REQUIRED_AND_PREFERRED_ATTENDANCE_CONFLICT, PREFERRED_ATTENDANCE_CONFLICT,
+
+                ROOM_CONFLICT, DONT_GO_IN_OVERTIME, REQUIRED_ATTENDANCE_CONFLICT, REQUIRED_ROOM_CAPACITY};
+            nextRow();
+            nextHeaderCell("");
+            writeTimeGrainDaysHeaders();
+            nextRow();
+            nextHeaderCell("Room");
+            writeTimeGrainHoursHeaders();
+            for (Room room : solution.getRoomList()) {
+                nextRow();
+                currentRow.setHeightInPoints(2 * currentSheet.getDefaultRowHeightInPoints());
+                nextCell().setCellValue(room.getName());
+                List<MeetingAssignment> roomMeetingAssignmentList = solution.getMeetingAssignmentList().stream()
+                    .filter(meetingAssignment -> meetingAssignment.getRoom() == room).collect(toList());
+
+                List<MeetingAssignment> mergePreviousMeetingList = null;
```

Comment:
Looks like this variable is only used as a flag. The list seem to never be modified or read. Instead, the variable is only switching between null and non-null states and being null-checked.